pipelines/curate_agentic.py

```
@@ -638,6 +789,14 @@ def _write_new_json(path: Path, value: Any) -> None:
 
 def write_cleaned_tree(run: dict[str, Any], out: Path) -> None:
     """Write retained JSONL plus scanner-safe JSON metadata in a new directory."""
+    mill_summary = run.get("summary", {}).get("mill_family")
+    if (
+        isinstance(mill_summary, dict)
+        and mill_summary.get("quarantine_applied") is False
+    ):
+        raise ValueError(
+            "refusing cleaned output without multi-factory mill ownership context"
+        )
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)